emission/storage/timeseries/builtin_timeseries.py

```diff
@@ -258,11 +255,19 @@ def _get_entries_for_timeseries(self, tsdb, key_list, time_query, geo_query,
             # Out[593]: 449869
             ts_db_result.limit(edb.result_limit)
         else:
-            ts_db_result = tsdb.find(INVALID_QUERY).hint([("metadata.key", 1)])
-            ts_db_count = 0
+            ts_db_result = []
+
+        # This list() conversion causes the cursor to be consumed, which is memory-expensive,
+        # but faster as it allows us to get the count without having to make an additional
+        # "count_documents" DB call.
+        # In a future situation where memory is constrained or DB calls are less expensive,
+        # we could skip the list() conversion and return the cursor directly.
+        # https://github.com/e-mission/e-mission-server/pull/1032#issuecomment-2685846702
+        ts_db_entries = list(ts_db_result)
+        ts_db_count = len(ts_db_entries)
```

Comment:
We can move this into the `if` block above since the `else` already has a list and a 0 count